html_ui/Pages/Api/NXApi.js

```diff
@@ -0,0 +1,221 @@
+class NXApi {
+    static getMetar(icao, source) {
+        if (!icao) {
+            throw ("No ICAO provided");
+        }
+
+        return fetch(`${NXApi.url}/metar/${icao}?source=${source}`)
+            .then((response) => {
+                if (!response.ok) {
+                    throw (response);
+                }
+
+                return response.json();
+            });
+    }
+
+    static getTaf(icao, source) {
+        if (!icao) {
+            throw ("No ICAO provided");
+        }
+
+        return fetch(`${NXApi.url}/taf/${icao}?source=${source}`)
+            .then((response) => {
+                if (!response.ok) {
+                    throw (response);
+                }
+
+                return response.json();
+            });
+    }
+
+    static getAtis(icao, source) {
+        if (!icao) {
+            throw ("No ICAO provided");
+        }
+
+        return fetch(`${NXApi.url}/atis/${icao}?source=${source}`)
+            .then((response) => {
+                if (!response.ok) {
+                    throw (response);
+                }
+
+                return response.json();
+            });
+    }
+
+    static connectTelex(flightNo) {
+        // TELEX disabled
+        if (NXDataStore.get("CONFIG_ONLINE_FEATURES_STATUS", "DISABLED") !== "ENABLED") {
+            return Promise.reject(NXApi.disabledError);
+        }
+
+        if (!flightNo) {
+            return Promise.reject(NXApi.disabledError);
+        }
+
+        const connectBody = NXApi.buildTelexBody(flightNo);
+        const headers = {"Content-Type": "application/json"};
+
+        return fetch(`${NXApi.url}/txcxn`, {method: "POST", body: JSON.stringify(connectBody), headers})
+            .then((response) => {
+                if (!response.ok) {
+                    throw (response);
+                }
+
+                return response.json()
+                    .then((data) => {
+                        NXApi.accessToken = data.accessToken;
+                        return data;
+                    });
+            });
+    }
```

Comment:
Do we need to have these telex functions present? Since we're only using the API for metar and atis requests, it would probably be best to only include functions for fetching what we actually need from the FBW API.

Reply:
I would like to implement these functions in the future, but for now i can remove them.